### ISSUE TITLE
What if … Cross posts were only symlinks?

### ISSUE THREAD

**Issue by kromonos** [LemmyNet/lemmy#2012]
Parallel to the suggestion I already wrote at https://fapsi.be/post/18979, the idea that a cross-post is less a separate post and more a link to the original might be worth considering implementing?  
It would have the advantage that discussions on one and the same topic would not be spread over several threads but could be communicated directly with each other.

**Comment by Tmpod** [comment]
That may or may not be desired; depends on the community. I think having the ability to "symlink" a post would be really cool, but at the same time, I feel there should be a way to have the current behavior.

**Comment by kromonos** [comment]
Maybe, if one would set the “symlink” as default behavior, but add a button like “Create new post” with renaming the “Create” to “Link”?
![Untitled](https://user-images.githubusercontent.com/1553902/147078265-8ecbc927-636a-41f4-8fdd-348739810ff1.png)

**Comment by Tmpod** [comment]
That sounds good to me

**Comment by dessalines** [comment]
Does symlink mean the url would go to the original lemmy post? I'm confused about how this would work.

Each post in a community has its own comment section and stands on its own, which is how reddit does this. I then made a *URL Searcher* that finds historical / other posts with that same URL to find historical posts or cross-posts in other communities.

A symlink doesn't make sense to me, because that means it wouldn't even really be a post in a different community, since it wouldn't have a comment section and just go to the first community.

**Comment by Dakkaron** [comment]
Yeah, I think that would be the point.

If post 2 symlinks to post 1, and you open post 2, it would just redirect you to post 1. (Or maybe instead of using redirection, it would just load everything as if you opened post 1.)

The point would be that you can notify a different community about an interesting post without splitting the discussion.

So say you have a post that is posted on feddit.de/c/Vienna, but that's also interesting to /c/Austria (because Vienna is in Austria), then you don't have to split the discussion between two small user bases, but instead both communities would comment on the same thread.

**Comment by Nutomic** [comment]
You can already do this by putting the link of the original post into the url field when writing a new post. So this would only require changing the behaviour of the crosspost button in lemmy-ui.

**Comment by chaorace** [comment]
> You can already do this by putting the link of the original post into the url field when writing a new post. So this would only require changing the behaviour of the crosspost button in lemmy-ui, no backend changes.

@Nutomic: I don't believe that we can already do this. Currently, comments crossposted in this way have separated comment sections ([link A](https://lemmy.sdf.org/post/501442), [link B](https://lemmy.sdf.org/post/501455)). What's being requested here is for the comment section of the "child" post to effectively be a reference to the "parent" post's comment section.

![1688395673](https://github.com/LemmyNet/lemmy/assets/6462592/073e5d5f-94ad-461e-8129-e17d8cfc131e)

Yes, this is _theoretically_ something that can be achieved on the ui-side by simply rewiring the comment link displayed under the "child" post to point to the "parent" post's comment section, but there are two major issues with this:

1. If a distinct comment section still exists for the "child" post, there will be inconsistencies between client implementations/versions, as it's non-obvious that the "child" post's comment section only exists in a vestigial capacity
2. Not all communities will want to deal with the hassle of posts where the comment section exists outside of their jurisdiction. Making "child" posts distinct from the existing crosspost mechanism will save headaches in the future when it comes to implementing more fine-grained community controls

At a minimum, these "child" posts could simply have "dummy" comment sections -- basically an API black-hole which rejects all attempts to post comments. This will force client implementations to properly implement the expected behavior of linking to the "parent" post's comment section.

If possible, though, I think it would be way better to instead do this at the DB-level so that "child" posts in place of their own comment section instead store a reference to the "parent" post's comment section. Perhaps diesel has some sort of alias/symlink-like capability that could theoretically make this more trivial?

---

Regardless of the implementation, allow me to advocate for this feature by explaining my own use-case:

I'm currently porting an anime episode-discussion bot called [Holo](https://github.com/r-anime/holo/pull/169) to Lemmy. It's designed to facilitate discussion threads like the kind you see [here](https://old.reddit.com/r/anime/comments/14lc018/oshi_no_ko_episode_11_discussion_final/). Weekly threads like these are the lifeblood of communities surrounding serialized media including anime, television, and comics, but they require a critical mass of users to properly flourish.

In other words: most people won't talk into an empty room. Regardless of how many crossposts I have the bot make a cross different anime communities, discussion will inevitably flock to one single thread -- probably whichever one happens to receive the first comment. In other words: these threads are a major cause of community centralization. It puts me as a developer in a very awkward position, because I want *multiple* anime communities to flourish. I don't want to smother partner communities in the crib because of a first-mover advantage!
